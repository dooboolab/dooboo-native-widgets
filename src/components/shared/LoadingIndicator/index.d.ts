import * as React from 'react';
import { ViewStyle, ImageSourcePropType } from 'react-native';

interface Props {
  containerStyle?: ViewStyle,
  style?: ViewStyle,
  color?: string;
  size?: number | 'small' | 'large';
}

const LoadingIndicator: React.SFC<Props> = (props) => {

};

export default LoadingIndicator;
