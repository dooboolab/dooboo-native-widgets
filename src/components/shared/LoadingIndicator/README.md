# LoadingIndicator
> Simple drop down item component for react-native. This component supports drop down toggle with animation.<br/>
<img src="https://user-images.githubusercontent.com/27461460/47951961-a7a7e500-dfab-11e8-9189-86c0eddb6e12.gif"/>

  containerStyle?: ViewStyle,
  style?: ViewStyle,
  color?: string;
  size?: number | 'small' | 'large';
  children?: any;

## Props
|              | necessary | types     | default                      |
|--------------|-----------|-----------|------------------------------|
|containerStyle|           | ViewStyle |                              |
|style         |           | ViewStyle |                              |
|color         |           | string    |                              |
|size          |           | string    | `number | 'small' | 'large'` |

## Getting started

* Import
  ```javascript
  import { LoadingIndicator } from 'dooboo-native-widgets';
  ```

* Usage
  ```javascript
  function Page(props: Props) {
    return (
      <LoadingIndicator />
    );
  }
  ```
