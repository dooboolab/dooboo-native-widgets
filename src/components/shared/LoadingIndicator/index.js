import React, { Component } from 'react';
import {
  ActivityIndicator,
  StyleSheet,
  View,
  ViewPropTypes,
} from 'react-native';
import PropTypes from 'prop-types';

const styles = StyleSheet.create({
  container: {
    flex: 1,
    position: 'absolute',
    backgroundColor: 'transparent',
    flexDirection: 'row',
    alignItems: 'center',
    alignSelf: 'center',
    justifyContent: 'center',
    height: '100%',
    width: '100%',
  },
});

function LoadingIndicator(props) {
  return (
    <View style={[
      styles.container,
      props.containerStyle,
    ]}>
      <ActivityIndicator
        style={props.style}
        size={props.size}
        color={props.color}
      />
    </View>
  );
}

LoadingIndicator.defaultProps = {
  size: 'large',
  color: '#969696',
};

LoadingIndicator.propTypes = {
  containerStyle: ViewPropTypes.style,
  style: ViewPropTypes.style,
  color: PropTypes.string,
  size: PropTypes.oneOfType([
    PropTypes.number,
    PropTypes.oneOf(['small', 'large']),
  ]),
};

export default LoadingIndicator;
