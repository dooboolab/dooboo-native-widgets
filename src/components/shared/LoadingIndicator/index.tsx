import { ActivityIndicator, StyleSheet, View, ViewStyle } from 'react-native';

import React from 'react';

const styles = StyleSheet.create({
  container: {
    flex: 1,
    position: 'absolute',
    backgroundColor: 'transparent',
    flexDirection: 'row',
    alignItems: 'center',
    alignSelf: 'center',
    justifyContent: 'center',
    height: '100%',
    width: '100%',
  },
});

interface Props {
  containerStyle?: ViewStyle;
  style?: ViewStyle;
  color?: string;
  size?: number | 'small' | 'large';
}

function LoadingIndicator(props: Props) {
  return (
    <View style={[styles.container, props.containerStyle]}>
      <ActivityIndicator
        style={props.style}
        size={props.size}
        color={props.color}
      />
    </View>
  );
}

LoadingIndicator.defaultProps = {
  size: 'large',
  color: '#969696',
};

export default LoadingIndicator;
