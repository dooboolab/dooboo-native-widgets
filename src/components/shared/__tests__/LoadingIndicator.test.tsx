import 'react-native';

import * as React from 'react';

import { RenderResult, fireEvent, render } from '@testing-library/react-native';

import LoadingIndicator from '../LoadingIndicator';
// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';

let props: any;
let component: React.ReactElement;
let testingLib: RenderResult;

const createTestProps = (obj: object): object => ({
  navigation: {
    navigate: jest.fn(),
  },
  ...obj,
});

describe('[LoadingIndicator] render', () => {
  beforeEach(() => {
    props = createTestProps({});
    component = <LoadingIndicator {...props} />;
  });

  it('renders without crashing', () => {
    const rendered: renderer.ReactTestRendererJSON = renderer
      .create(component)
      .toJSON();
    expect(rendered).toMatchSnapshot();
    expect(rendered).toBeTruthy();
  });
});
