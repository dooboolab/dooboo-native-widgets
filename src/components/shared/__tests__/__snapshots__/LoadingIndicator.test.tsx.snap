// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[LoadingIndicator] render renders with !size with none string imgSource 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "alignSelf": "center",
        "backgroundColor": "transparent",
        "flex": 1,
        "flexDirection": "row",
        "height": "100%",
        "justifyContent": "center",
        "position": "absolute",
        "width": "100%",
      },
      undefined,
    ]
  }
>
  <Image
    source={10}
    style={Object {}}
  />
</View>
`;

exports[`[LoadingIndicator] render renders with !size with string imgSource 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "alignSelf": "center",
        "backgroundColor": "transparent",
        "flex": 1,
        "flexDirection": "row",
        "height": "100%",
        "justifyContent": "center",
        "position": "absolute",
        "width": "100%",
      },
      undefined,
    ]
  }
>
  <Image
    source={
      Object {
        "uri": "test",
      }
    }
    style={Object {}}
  />
</View>
`;

exports[`[LoadingIndicator] render renders with [imgSource] 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "alignSelf": "center",
        "backgroundColor": "transparent",
        "flex": 1,
        "flexDirection": "row",
        "height": "100%",
        "justifyContent": "center",
        "position": "absolute",
        "width": "100%",
      },
      undefined,
    ]
  }
>
  <Image
    source={
      Object {
        "uri": "http",
      }
    }
    style={
      Object {
        "height": 80,
        "width": 80,
      }
    }
  />
</View>
`;

exports[`[LoadingIndicator] render renders with size=small 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "alignSelf": "center",
        "backgroundColor": "transparent",
        "flex": 1,
        "flexDirection": "row",
        "height": "100%",
        "justifyContent": "center",
        "position": "absolute",
        "width": "100%",
      },
      undefined,
    ]
  }
>
  <Image
    source={
      Object {
        "uri": "http",
      }
    }
    style={
      Object {
        "height": 50,
        "width": 50,
      }
    }
  />
</View>
`;

exports[`[LoadingIndicator] render renders with size=test 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "alignSelf": "center",
        "backgroundColor": "transparent",
        "flex": 1,
        "flexDirection": "row",
        "height": "100%",
        "justifyContent": "center",
        "position": "absolute",
        "width": "100%",
      },
      undefined,
    ]
  }
>
  <Image
    source={10}
    style={
      Object {
        "height": "test",
        "width": "test",
      }
    }
  />
</View>
`;

exports[`[LoadingIndicator] render renders with size=undefined 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "alignSelf": "center",
        "backgroundColor": "transparent",
        "flex": 1,
        "flexDirection": "row",
        "height": "100%",
        "justifyContent": "center",
        "position": "absolute",
        "width": "100%",
      },
      undefined,
    ]
  }
>
  <Image
    source={
      Object {
        "uri": "http",
      }
    }
    style={
      Object {
        "height": 80,
        "width": 80,
      }
    }
  />
</View>
`;

exports[`[LoadingIndicator] render renders without crashing 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "alignSelf": "center",
        "backgroundColor": "transparent",
        "flex": 1,
        "flexDirection": "row",
        "height": "100%",
        "justifyContent": "center",
        "position": "absolute",
        "width": "100%",
      },
      undefined,
    ]
  }
>
  <ActivityIndicator
    animating={true}
    color="#969696"
    hidesWhenStopped={true}
    size="large"
  />
</View>
`;
